cmd/bootstrap/cmd/finalize.go

```diff
@@ -170,12 +170,15 @@ func finalize(cmd *cobra.Command, args []string) {
 	constructRootResultAndSeal(flagRootCommit, block, stakingNodes, assignments, clusterQCs, dkgData)
 	log.Info().Msg("")
 
-	log.Info().Msg("copying internal private keys to output folder")
-	err := copyDir(flagInternalNodePrivInfoDir, filepath.Join(flagOutdir, model.DirPrivateRoot))
-	if err != nil {
-		log.Error().Err(err).Msg("could not copy private key files")
+	// copy files only if the directories differ
+	if flagInternalNodePrivInfoDir != flagOutdir {
+		log.Info().Msg("copying internal private keys to output folder")
+		err := copyDir(flagInternalNodePrivInfoDir, filepath.Join(flagOutdir, model.DirPrivateRoot))
+		if err != nil {
+			log.Error().Err(err).Msg("could not copy private key files")
+		}
+		log.Info().Msg("")
 	}
```

Comment:
better to add an `else` statement to print that the folders are the same and skip the copying.